engine/baml-vm/src/bytecode.rs

```diff
@@ -180,13 +180,20 @@ pub struct Bytecode {
 
     /// Constant pool.
     pub constants: Vec<Value>,
+
+    /// Source line mapping.
+    ///
+    /// Maps instruction indices to their source line numbers.
+    /// Each element corresponds to an instruction at the same index.
+    pub source_lines: Vec<Option<usize>>,
```

Comment:
Why is it `Option<usize>`? Can a bytecode instruction not map to any line in the source? Looks like this is never set to `None` either way.

Reply:
It turns out even a 0 is fine since line numbers always begin with `1`
Resolved via https://github.com/BoundaryML/baml/pull/2132/commits/d0aa17a6d62106758b74fa242e56cb87a1ac18ea